### ISSUE TITLE
Implement solaris.nets model checkpointing and callbacks

### ISSUE THREAD

**Issue by nrweir** [CosmiQ/solaris#39]
@nrweir commented on [Tue Mar 05 2019](https://github.com/CosmiQ/cw-nets/issues/40)

# Implement model checkpointing

We currently don't have model checkpoints or callbacks implemented in this library. We need a way to:

1. save models during training (the "best" loss during training, as well as the final version)
2. Implement Keras callbacks (`ReduceLRonPlateau`, etc.)
3. Implement equivalent callback functionality for PyTorch models

**Comment by nrweir** [comment]
Resolved with #62
